src/java.base/share/classes/sun/net/www/protocol/jrt/JavaRuntimeURLConnection.java

```diff
@@ -45,99 +44,79 @@
  */
 public class JavaRuntimeURLConnection extends URLConnection {
 
-    // ImageReader to access resources in jimage
-    private static final ImageReader reader = ImageReaderFactory.getImageReader();
-
-    // the module and resource name in the URL
+    // ImageReader to access resources in jimage (never null).
+    private static final ImageReader READER = ImageReaderFactory.getImageReader();
+
+    // The module and resource name in the URL ("jrt:/<module-name>/<resource-name>").
+    //
+    // It is important to note that all of this information comes from the given
+    // URL's path part, and there's no requirement for there to be distinct rules
+    // about percent encoding, and it is likely that any differences between how
+    // module names and resource names are treated is unintentional. The rules
+    // about percent encoding may well be tightened up in the future.
```

Comment:
I think it would be better to drop this paragraph from the comment, it just begs too many questions.